### ISSUE TITLE
Bot Exiting Mid-search/Mid-game

### ISSUE THREAD

**Issue by Kyrluckechuck** [TeamFightTacticsBots/Alune#95]
## Description

Hola folks!

Not sure what's changed in recent updates here that is affecting this, but something about the game loop appears to be exiting early with my setup. Currently running from latest `main`, with the attached config & log (with `DEBUG` logging enabled).
It looks like with the `screen_record` `enabled: true` flag set, it's encountering some sort of error and just aborting/exiting without any log to indicate why?

Basically I can start up the bot and it will nearly always immediately quit either after starting the search and before hitting accept, or just mid-game at some point.

## Extra
[exit-early.zip](https://github.com/user-attachments/files/18089079/exit-early.zip)

I've just reverted `enabled: false` for the `screen_record` setting and currently running through, will comment on this issue if it "resolves" it.

**Comment by akshualy** [comment]
Hi, the attached logs seem to not be from a session where you had screen record enabled, as it being enabled is logged. Could you check your other log files?

**Comment by Kyrluckechuck** [comment]
Odd, I thought it had been -- this one specifically mentions it!
[2024-12-10_22-14-13_979225.log](https://github.com/user-attachments/files/18096654/2024-12-10_22-14-13_979225.log)

Also just confirming that disabling it worked as intended and ran overnight, so looks to be something specific to the recording flow

**Comment by akshualy** [comment]
Thanks for reporting this, it's a little curious how nothing is logged because we have a broad exception catch for this exact reason.
I'll look into the reasons, if the process just got killed it might also be a performance related thing since screen recording usually takes a larger toll on system resources.

**Comment by Kyrluckechuck** [comment]
No worries -- it looks to be consistent for what that's worth, and I don't _think_ it's related to high system resources but could very well be possible
